File: backend/app/types/gl_accounts.py
from pydantic import BaseModel
from fastapi import Query
from typing import Optional
from enum import Enum

class AccountType(str, Enum):
	asset = "Asset"
	liability = "Liability"
	equity = "Equity"
	revenue = "Revenue"
	expense = "Expense"

class NormalBalance(str, Enum):
	debit = "debit"
	credit = "credit"

class CreateGLAccountRequest(BaseModel):
	account_number: str
	name: str
	account_type: AccountType
	normal_balance: NormalBalance
	description: Optional[str] = None

class UpdateGLAccountRequest(BaseModel):
	account_number: Optional[str] = None
	name: Optional[str] = None
	account_type: Optional[AccountType] = None
	normal_balance: Optional[NormalBalance] = None
	description: Optional[str] = None
	is_active: Optional[bool] = None

class GLAccountFilters(BaseModel):
	account_number: Optional[str] = Query(default=None)
	name: Optional[str] = Query(default=None)
	account_type: Optional[str] = Query(default=None)
	is_active: Optional[bool] = Query(default=None)

File: frontend/app/lib/types/gl_accounts.ts
export interface GLAccount {
	id: number;
	account_number: string;
	name: string;
	account_type: string;
	normal_balance: "debit" | "credit";
	description: string | null;
	is_active: boolean;
}

export interface GLAccountCreate {
	account_number: string;
	name: string;
	account_type: string;
	normal_balance: "debit" | "credit";
	description?: string;
}

export interface GLAccountPatch {
	account_number?: string;
	name?: string;
	account_type?: string;
	normal_balance?: "debit" | "credit";
	description?: string;
	is_active?: boolean;
}
